src/nimble.nim

```diff
@@ -1043,6 +1043,56 @@ proc check(options: Options) =
     display("Failure:", "Validation failed", Error, HighPriority)
     quit(QuitFailure)
 
+
+proc sample(options: Options) =
+  ## ~ copies sample files to current directory - does not creates new subdirectory.
+  ##   (copyDir nimblepath/pkgs/pkg/samples/arg to current directory.)
+  ## ~ finds `arg` in subdirectories. "nimble sample pkgname sub"
+  ## ~ looks for pkg version if requested. "nimble sample pkgname '0.26.2' demo1"
+  ## ~ copies the first match found
+  ## ~ overwrites existing files without question
+
+  let installedPkgs = getInstalledPkgsMin(options.getPkgsDir(), options)
+  var version = toVersionRange(options.action.ver)
+
+  for (pkg,meta) in installedPkgs: # pkg exists?
+    if pkg.name == options.action.package and
+      (withinRange(pkg, version) or pkg.version == "#head"): #"#head" == develop 
+        var (dir, name, ext) = splitFile(pkg.myPath)
+        let examplesDir = dir & DirSep & "samples"
```

Comment:
This directory won't be copied in there. You'll need to implement this installation logic for examples as well (`examples/` should probably go to `~/.nimble/examples/pkg_name-ver`).

Reply:
(hopefully) we will get back to this, as it is not clear for me, but let me first fix the other issues and add some explanation. thank you!

Reply:
Might be easier to create a separate PR for this. We can iterate faster that way.

Reply:
You will need this as well to test this PR properly. I'm not sure how you're testing it currently as I don't see how the `examples` can be installed right now.

Reply:
yes, examples/ lives in ~/.nimble/pkgs/pkg, just like tests/.
the idea is, that the package maintainer should/could add it, like tests/.
wether they using it or not, is up for the creator.

the main difference from eg tasks is, that i don't have to be in the packages directory,
i call nimble examples from my source dir. (so i have to lookup for pkgDir)

explanation:
i have a project,, the problem i face is that: to get a snippet from the pkgdir/examples/
i have to get the pkgdir, wich is not so easy,, except, if i hack nimble itself - and here we go ;)
...so in the end, i could publish, and write: 
"create a dir test, or whatever. type: nimble example myproj helloworld, then type nim c -r helloworld"

Yess, it is for convenience...  ;)
(((in defense, if needed: if a kid moves the example form gamengineX via drag and drop, then the kid will cry next time)))

((Other idea was, to extend tasks: nimble run pkgname taskname
wich would get currentDir and pkgDir.))

thank you for your time!

Reply:
> You will need this as well to test this PR properly. I'm not sure how you're testing it currently as I don't see how the `examples` can be installed right now.

All right, i will think about how to write tests for it.
As examples/ should be added by the pkg writer, same way as tests/ - i sleep on it a little.
Thank again!